Netable/Netable/Interceptor.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  Interceptor.swift
+//  Netable
+//
+//  Created by Brendan Lensink on 2023-06-06.
+//  Copyright © 2023 Steamclock Software. All rights reserved.
+//
+
+
+import Foundation
+
+public struct Interceptor: Sendable {
```

Comment:
Minor naming nitpick, but this being called "Interceptor" seems a bit weird, my natural inclination would be to assume these two types are both singular and on is like a refinement of the other, not that "Interceptor" is a collection of "RequestInterceptors"

I feel like "Interceptor" should be called like "InterceptorList" or something more obviously a container.